src/HDF5.jl

```diff
@@ -85,13 +84,13 @@ hdf5_type_id(::Type{Int64})      = H5T_NATIVE_INT64
 hdf5_type_id(::Type{UInt64})     = H5T_NATIVE_UINT64
 hdf5_type_id(::Type{Float32})    = H5T_NATIVE_FLOAT
 hdf5_type_id(::Type{Float64})    = H5T_NATIVE_DOUBLE
-hdf5_type_id(::Type{HDF5ReferenceObj}) = H5T_STD_REF_OBJ
+hdf5_type_id(::Type{Reference})  = H5T_STD_REF_OBJ
 
 hdf5_type_id(::Type{S}) where {S<:AbstractString}  = H5T_C_S1
 hdf5_type_id(::Type{C}) where {C<:CharType}        = H5T_C_S1
 
 const HDF5BitsKind = Union{Bool, Int8, UInt8, Int16, UInt16, Int32, UInt32, Int64, UInt64, Float32, Float64}
```

Comment:
These are used internally, but could also remove the `HDF5` prefix.

Reply:
There's a couple of others like this including `HDF5Scalar` that are internally used.

Reply:
@jmert  Should I save these changes in a new PR ?

Reply:
I'll do them in a separate PR since they are more internal utility constants, rather than user-facing ones, so logically it would make sense to be a separate PR.